core/procs.go

```diff
@@ -1870,6 +1871,40 @@ var procIsNamespaceInitialized = func(args []Object) Object {
 	return MakeBoolean(found && ns.Lazy == nil)
 }
 
+var procPrecision = func(args []Object) Object {
+	prec := int64(-1)
+	switch n := args[0].(type) {
+	case Number:
+		switch n := n.(type) {
+		case *BigInt:
+			prec = int64(n.b.BitLen())
+		case *BigFloat:
+			prec = int64(n.b.Prec())
+		case Int:
+			prec = int64(bits.UintSize - 1)
+		case Double:
+			prec = 53
+		}
+	}
+	return MakeBigInt(prec)
+}
+
+var procSetPrecision = func(args []Object) Object {
+	p := EnsureArgIsNumber(args, 0).Int().I
+	if p < 0 {
+		panic(RT.NewArgTypeError(0, args[0], "non-negative Int"))
+	}
+	prec := uint(p)
+	switch n := args[1].(type) {
+	case Number:
+		switch n := n.(type) {
+		case *BigFloat:
```

Comment:
The same here, why double switching on type? Perhaps `EnsureArgIsBigFloat` and `ExtractBigFloat` need to be created, similar to helpers for other types?

Reply:
Actually, `EnsureArgIsBigFloat` is auto-generated...

Reply:
Yup, that's an easy fix.